### PR TITLE
feat(apigatewayv2-integrations): http private integrations - tls config

### LINKED ISSUE #14036: (apigatewayv2-integrations): Set SecureServerName with higher level HTTP API libraries
I'd like to have the ability to set the SecureServerName for HTTP APIs with L2 or L3 constructs. Setting the SecureServerName through Cloudformation can be done through `AWS::ApiGatewayV2::Integration`'s property `TlsConfig`.

### Use Case

Functionality only exists in L1 construct, would like to be able to set SecureServerName using higher level libraries

### Proposed Solution

_tbd_

### Other

_Sourced from guidance issue_ #13673 

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by dan-lind**
Adds the option to specify tlsConfig for `AWS::ApiGatewayV2::Integration` resources, to enable use of HTTPS when configuring private integrations.

closes #14036

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by dan-lind** [comment]
@peterwoodworth I opted to update a unit test in `aws-apigatewayv2/test/http/route.test.ts` which tested the existing L2 construct. Also added separate tests for the L3 constructs, hope that is fine. 

Not entirely clear on why the build failed, works just fine locally. 
This looks suspicious though.
```
Server:
ERROR: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

**Comment by peterwoodworth** [comment]
Where exactly are you getting that build failure?

**Comment by dan-lind** [comment]
> Where exactly are you getting that build failure?

Check the CodeBuild CI Report above, it says "Result: Failed"

**Comment by peterwoodworth** [comment]
Oh duh 🤦 right there. I'm not sure why that happened, no idea why you'd get that error there 🤷

**Review by peterwoodworth — COMMENTED** [review]
This looks like it works fine. However we could afford to have more abstraction here, right? Why not allow the user to input the property `SecureServerName` and then have the CDK handle setting the TlsConfig -> ServerNameToVerify property?

**Comment by peterwoodworth** [comment]
We can deal with the build errors once we're ready to approve. Hopefully it won't happen again but if it does we can figure out how to deal with that

**Comment by dan-lind** [comment]
Agree! I've updated the MR per your suggestion. Let's see if the build passes this time.

**Comment by peterwoodworth** [comment]
Thank you! It looks good to me, I'll leave it to @nija-at for final review

File: packages/@aws-cdk/aws-apigatewayv2/lib/http/integration.ts
Comment (nija-at) [inline packages/@aws-cdk/aws-apigatewayv2/lib/http/integration.ts:128]:
```suggestion
   * @default undefined private integration traffic will use HTTP protocol
```

File: packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/base-types.ts
Comment (nija-at) [inline packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/base-types.ts:23]:
```suggestion
  * @default undefined private integration traffic will use HTTP protocol
```

File: packages/@aws-cdk/aws-apigatewayv2/lib/http/integration.ts
Comment (nija-at) [inline packages/@aws-cdk/aws-apigatewayv2/lib/http/integration.ts:237]:
```suggestion
   * @default undefined private integration traffic will use HTTP protocol
```

**Review by nija-at — DISMISSED** [review]
Nice 💯

**Review by nija-at — APPROVED** [review]

**Pull request merged**
